test/lvm/lvm_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+//go:build !windows
+
+package acceptance
+
+import (
+	"log"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+)
+
+const (
+	namespace    = "LVMTest"
+	agentRuntime = 3 * time.Minute
+)
+
+var envMetaDataStrings = &(environment.MetaDataStrings{})
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
+}
+
+type LVMTestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+func (t LVMTestRunner) Validate() status.TestGroupResult {
+	metricsToFetch := t.GetMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		testResults[i] = t.validateDiskMetric(metricName)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.GetTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *LVMTestRunner) validateDiskMetric(metricName string) status.TestResult {
+	testResult := status.TestResult{
+		Name:   metricName,
+		Status: status.FAILED,
+	}
+
+	dims, failed := t.DimensionFactory.GetDimensions([]dimension.Instruction{
+		{
+			Key:   "InstanceId",
+			Value: dimension.UnknownDimensionValue(),
+		},
+	})
+
+	if len(failed) > 0 {
+		return testResult
+	}
+
+	fetcher := metric.MetricValueFetcher{}
+	values, err := fetcher.Fetch(namespace, metricName, dims, metric.AVERAGE, metric.HighResolutionStatPeriod)
+
+	log.Printf("metric values are %v", values)
+	if err != nil {
+		return testResult
+	}
+
+	if !metric.IsAllValuesGreaterThanOrEqualToExpectedValue(metricName, values, 0) {
+		return testResult
+	}
+
+	testResult.Status = status.SUCCESSFUL
+	return testResult
+}
+
+func (t LVMTestRunner) GetTestName() string {
+	return namespace
+}
+
+func (t LVMTestRunner) GetAgentConfigFileName() string {
+	return "config.json"
+}
+
+func (t LVMTestRunner) GetMeasuredMetrics() []string {
+	return []string{"disk_free", "disk_used", "disk_total"}
+}
+
+func (t *LVMTestRunner) SetupBeforeAgentRun() error {
```

Comment:
Does this need to also set up the config for the agent now?

Reply:
It is done by another function `runAgent`